### PR TITLE
Full-text search support for SmartStore on Android

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "external/shared"]
 	path = external/shared
-        url = https://github.com/forcedotcom/SalesforceMobileSDK-Shared.git
```
Comment (wmathurin) [inline .gitmodules]: Will re-point to wmathurin once I wrap up the FTS work for hybrid

File: external/sqlcipher/SALESFORCE_README [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Binaries in this folder were obtained from: https://www.zetetic.net/sqlcipher/open-source/
```
Comment (wmathurin) [inline external/sqlcipher/SALESFORCE_README]: Readme explaining how to build the .so files with FTS enabled

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/phonegap/SmartStorePlugin.java [hunk 3]
```diff
@@ -65,6 +65,7 @@
 	public static final String LIKE_KEY = "likeKey";
 	public static final String MATCH_KEY = "matchKey";
 	public static final String SMART_SQL = "smartSql";
+	public static final String ORDER_PATH = "orderPath";
```
Comment (wmathurin) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/phonegap/SmartStorePlugin.java]: Previously you could only order by the field you were searching by.
But for full-text search, you can search across all indexed fields, and it didn't make sense not to have a sorting.
For backward compatibility, the javascript code uses indexPath as the orderPath when no orderPath is provided.

File: libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/AbstractSmartStoreTest.java [hunk 4]
```diff
@@ -888,345 +887,4 @@ public void testGetDatabaseSize() throws JSONException {
 		assertTrue("Database should be larger now", store.getDatabaseSize() > initialSize);
 	}
 	
-	/**
-	 * Test for getSoupIndexSpecs
-	 * 
-	 * @throws JSONException
-	 */
-	public void testGetSoupIndexSpecs() throws JSONException {
```
Comment (wmathurin) [inline libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/AbstractSmartStoreTest.java]: Those tests have moved to SmartStoreAlterTest.java

File: libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreAlterTest.java [hunk 5]
```diff
@@ -0,0 +1,560 @@
...
+        assertEquals("San Francisco should be first", "San Francisco", results2.getJSONObject(0).get("name"));
+        assertEquals("Paris should be first", "Paris", results2.getJSONObject(1).get("name"));
+    }
+
+    /**
+     * Test for alterSoup with column type change from string to full_text
+     *
+     * throws JSONException
+     */
+    public void testAlterSoupTypeChangeStringToFullText() throws JSONException {
```
Comment (wmathurin) [inline libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreAlterTest.java]: New test for FTS

File: libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreAlterTest.java [hunk 6]
```diff
@@ -0,0 +1,560 @@
...
+        // Checking db
+        checkDb(new long[]{elt1Id, elt2Id}, indexSpecsNew[0].type, indexSpecsNew[1].type);
+    }
+
+    /**
+     * Test for alterSoup with column type change from full_text to string
+     *
+     * throws JSONException
+     */
+    public void testAlterSoupTypeChangeFullTextToString() throws JSONException {
```
Comment (wmathurin) [inline libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreAlterTest.java]: New test for FTS

**Comment by bhariharan** [comment]
Looks good to me. :+1:

**Comment by wmathurin** [comment]
Now pointing to shared on forcedotcom

**Pull request merged**